modules/openxr/extensions/openxr_render_model_extension.cpp

```diff
@@ -0,0 +1,799 @@
+		// Obtain interaction ids
+		result = xrEnumerateInteractionRenderModelIdsEXT(session, &interaction_info, render_model_interaction_ids.size(), &interaction_count, render_model_interaction_ids.ptr());
+		if (XR_FAILED(result)) {
+			ERR_PRINT("OpenXR: Failed to obtain render model interaction ids [" + OpenXRAPI::get_singleton()->get_error_string(result) + "]");
+			return false;
+		}
+	}
+
+	// Remove render models that are no longer tracked
+	LocalVector<XrRenderModelIdEXT> erase_ids;
+	for (const KeyValue<XrRenderModelIdEXT, RID> &e : interaction_render_models) {
+		if (!render_model_interaction_ids.has(e.key)) {
+			if (e.value.is_valid()) {
+				render_model_destroy(e.value);
+			}
+
+			erase_ids.push_back(e.key);
+		}
+	}
+
+	// Remove these from our hashmap
+	for (const XrRenderModelIdEXT &id : erase_ids) {
+		interaction_render_models.erase(id);
+	}
+
+	// Now update our models
+	for (const XrRenderModelIdEXT &id : render_model_interaction_ids) {
+		// Q should we add a tracker wrapper for this? Might be nicer...
+		if (!interaction_render_models.has(id)) {
+			// Even if this fails we add it so we don't repeat trying to create it
+			interaction_render_models[id] = render_model_create(id);
+		}
+	}
+
+	return true;
+}
+
+bool OpenXRRenderModelExtension::has_render_model(RID p_render_model) const {
+	return render_model_owner.owns(p_render_model);
+}
+
+RID OpenXRRenderModelExtension::render_model_create(XrRenderModelIdEXT p_render_model_id) {
+	ERR_FAIL_COND_V_MSG(!render_model_ext, RID(), "Render model extension hasn't been enabled.");
+
+	OpenXRAPI *openxr_api = OpenXRAPI::get_singleton();
+	ERR_FAIL_NULL_V(openxr_api, RID());
+
+	XrSession session = openxr_api->get_session();
+	ERR_FAIL_COND_V(session == XR_NULL_HANDLE, RID());
+
+	RenderModel render_model;
+	render_model.xr_render_model_id = p_render_model_id;
+
+	// Start with the extensions that are supported in our base (see GLTFDocument::_parse_gltf_extensions).
+	Vector<const char *> supported_gltf_extensions = {
+		"KHR_lights_punctual",
+		"KHR_materials_pbrSpecularGlossiness",
+		"KHR_texture_transform",
+		"KHR_materials_unlit",
+		"KHR_materials_emissive_strength"
+	};
+
+	// Now find anything we support through plugins, which is a bit of a pain as they are converted to Strings
+	// and we need to convert them back.
+	Vector<CharString> char_extensions; // Just for temp storage of our c-strings.
+	Vector<Ref<GLTFDocumentExtension>> gltf_document_extensions = GLTFDocument::get_all_gltf_document_extensions();
+	for (Ref<GLTFDocumentExtension> &gltf_document_extension : gltf_document_extensions) {
+		Vector<String> supported_extensions = gltf_document_extension->get_supported_extensions();
+		for (String &extension : supported_extensions) {
+			char_extensions.push_back(extension.utf8());
+		}
+	}
+
+	// Now we can add them to our supported extensions list.
+	for (CharString &cs : char_extensions) {
+		supported_gltf_extensions.push_back(cs.get_data());
+	};
```

Comment:
Do we expect the gltf extensions to change during runtime? If not, we can populate the list once and just reuse it.

Reply:
Honestly, I don't know for sure. I assume we could load this once and re-use but in theory there could be an order of execution issue as I believe plugins can add support for extensions. 

That said, there isn't much overhead in this. It's not likely we would load more than a few individual models and the processing of the GLTF will far outweigh creating this list of extensions.